NullPointerException or ArrayIndexOutOfBoundsException launching sbt 1.5.0

## steps

1. Start `sbt`. Can be in an empty directory - doesn't matter. 

## problem

`NullPointerException` or `ArrayIndexOutOfBoundsException` on launch

```
$ /usr/local/bin/sbt
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at sbt.internal.XMainConfiguration.makeConfiguration(XMainConfiguration.java:102)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:45)
	at sbt.xMain.run(Main.scala:46)
	at xsbt.boot.Launch$.$anonfun$run$1(Launch.scala:149)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:176)
	at xsbt.boot.Launch$.run(Launch.scala:149)
	at xsbt.boot.Launch$.$anonfun$apply$1(Launch.scala:44)
	at xsbt.boot.Launch$.launch(Launch.scala:159)
	at xsbt.boot.Launch$.apply(Launch.scala:44)
	at xsbt.boot.Launch$.apply(Launch.scala:21)
	at xsbt.boot.Boot$.runImpl(Boot.scala:78)
	at xsbt.boot.Boot$.run(Boot.scala:73)
	at xsbt.boot.Boot$.main(Boot.scala:21)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.internal.XMainConfiguration.makeConfiguration(XMainConfiguration.java:84)
	... 13 more
Caused by: java.lang.NullPointerException
	at java.base/java.util.ArrayDeque.addLast(ArrayDeque.java:304)
	at java.base/java.util.ArrayDeque.add(ArrayDeque.java:495)
	at java.base/jdk.internal.loader.URLClassPath.<init>(URLClassPath.java:154)
	at java.base/jdk.internal.loader.URLClassPath.<init>(URLClassPath.java:171)
	at java.base/java.net.URLClassLoader.<init>(URLClassLoader.java:120)
	at sbt.internal.SbtInterfaceLoader.<init>(SbtInterfaceLoader.java:16)
	at sbt.internal.MetaBuildLoader.makeLoader(MetaBuildLoader.java:143)
	... 18 more
```

 The exception above occurs with SBT 1.5.0 on OpenJDK 15 and 11, and on  all JDKs when setting `sbt.version=1.4.9` via `project/build.properties`. With SBT 1.5.0 on OracleJDK 8 I get

```
$ JAVA_HOME=$(/usr/libexec/java_home -v 1.8.0) /usr/local/bin/sbt
java.lang.RuntimeException: java.lang.NullPointerException
	at sbt.internal.XMainConfiguration$ModifiedConfiguration.provider(XMainConfiguration.java:295)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:48)
	at sbt.xMain.run(Main.scala:46)
	at xsbt.boot.Launch$.$anonfun$run$1(Launch.scala:149)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:176)
	at xsbt.boot.Launch$.run(Launch.scala:149)
	at xsbt.boot.Launch$.$anonfun$apply$1(Launch.scala:44)
	at xsbt.boot.Launch$.launch(Launch.scala:159)
	at xsbt.boot.Launch$.apply(Launch.scala:44)
	at xsbt.boot.Launch$.apply(Launch.scala:21)
	at xsbt.boot.Boot$.runImpl(Boot.scala:78)
	at xsbt.boot.Boot$.run(Boot.scala:73)
	at xsbt.boot.Boot$.main(Boot.scala:21)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.NullPointerException
	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
	at sun.misc.URLClassPath.getLoader(URLClassPath.java:512)
	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:484)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:238)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:365)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:411)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at sbt.internal.XMainConfiguration$ModifiedConfiguration$ModifiedAppProvider$1$1.<init>(XMainConfiguration.java:132)
	at sbt.internal.XMainConfiguration$ModifiedConfiguration$ModifiedAppProvider$1.<init>(XMainConfiguration.java:128)
	at sbt.internal.XMainConfiguration$ModifiedConfiguration$ModifiedAppProvider.<init>(XMainConfiguration.java:126)
	at sbt.internal.XMainConfiguration$ModifiedConfiguration.provider(XMainConfiguration.java:293)
	... 13 more
```

When setting `sbt.version=1.3.13` via `project/build.properties` I get the following on all JDKs

```
$ /usr/local/bin/sbt
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.internal.XMainConfiguration.makeConfiguration(XMainConfiguration.scala:69)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:36)
	at sbt.xMain.run(Main.scala:39)
	at xsbt.boot.Launch$.$anonfun$run$1(Launch.scala:149)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:176)
	at xsbt.boot.Launch$.run(Launch.scala:149)
	at xsbt.boot.Launch$.$anonfun$apply$1(Launch.scala:44)
	at xsbt.boot.Launch$.launch(Launch.scala:159)
	at xsbt.boot.Launch$.apply(Launch.scala:44)
	at xsbt.boot.Launch$.apply(Launch.scala:21)
	at xsbt.boot.Boot$.runImpl(Boot.scala:78)
	at xsbt.boot.Boot$.run(Boot.scala:73)
	at xsbt.boot.Boot$.main(Boot.scala:21)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 82 out of bounds for length 82
	at sbt.internal.MetaBuildLoader.makeLoader(MetaBuildLoader.java:73)
	... 18 more
```

## expectation

SBT should launch

## notes

I run OS X 11.2.3 with SBT 1.5.0 installed via Homebrew, using `JAVA_HOME` to select various JDKs.

I have tried deleting `$HOME/.sbt/boot` as suggested on reddit, but the errors still occur.

In as much as I can gather from [their sbt.rb file](https://github.com/Homebrew/homebrew-core/blob/master/Formula/sbt.rb), Homebrew does not do much more than download and extract the tgz for you.

It does happen for me too with a fresh download

```
$ ht -F -d https://github.com/sbt/sbt/releases/download/v1.5.0/sbt-1.5.0.tgz -o sbt.tgz && tar xf sbt.tgz && ./sbt/bin/sbt
HTTP/2.0 200 OK
accept-ranges: bytes
age: 0
content-disposition: attachment; filename=sbt-1.5.0.tgz
content-length: 19985381
content-type: application/octet-stream
date: Tue, 06 Apr 2021 21:16:07 GMT
etag: "4a9fd0b28bd80bf22bb50a65602f4517"
last-modified: Sun, 04 Apr 2021 18:06:16 GMT
server: AmazonS3
strict-transport-security: max-age=31536000
via: 1.1 varnish, 1.1 varnish
x-amz-id-2: iZjxeNtSRziKPrEIAfkiIUenDqYWj3Rc8JJP47wRT+6iCzrysQ31zIm3f0TblnA8Q5Fd90ye0/w=
x-amz-request-id: KVK9GBXQ5MAN3V4A
x-cache: MISS, MISS
x-cache-hits: 0, 0
x-fastly-request-id: f9fdf57d6bb3a20d0e74727e1ad0de4fc690354a
x-served-by: cache-dca17755-DCA, cache-sea4425-SEA

Downloading 19.06MB to "sbt.tgz"
  [00:00:01] [########################################] 19.06MB 10.15MB/s ETA 0s
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at sbt.internal.XMainConfiguration.makeConfiguration(XMainConfiguration.java:102)
	at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:45)
	at sbt.xMain.run(Main.scala:46)
	at xsbt.boot.Launch$.$anonfun$run$1(Launch.scala:149)
	at xsbt.boot.Launch$.withContextLoader(Launch.scala:176)
	at xsbt.boot.Launch$.run(Launch.scala:149)
	at xsbt.boot.Launch$.$anonfun$apply$1(Launch.scala:44)
	at xsbt.boot.Launch$.launch(Launch.scala:159)
	at xsbt.boot.Launch$.apply(Launch.scala:44)
	at xsbt.boot.Launch$.apply(Launch.scala:21)
	at xsbt.boot.Boot$.runImpl(Boot.scala:78)
	at xsbt.boot.Boot$.run(Boot.scala:73)
	at xsbt.boot.Boot$.main(Boot.scala:21)
	at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.internal.XMainConfiguration.makeConfiguration(XMainConfiguration.java:84)
	... 13 more
Caused by: java.lang.NullPointerException
	at java.base/java.util.ArrayDeque.addLast(ArrayDeque.java:304)
	at java.base/java.util.ArrayDeque.add(ArrayDeque.java:495)
	at java.base/jdk.internal.loader.URLClassPath.<init>(URLClassPath.java:154)
	at java.base/jdk.internal.loader.URLClassPath.<init>(URLClassPath.java:171)
	at java.base/java.net.URLClassLoader.<init>(URLClassLoader.java:120)
	at sbt.internal.SbtInterfaceLoader.<init>(SbtInterfaceLoader.java:16)
	at sbt.internal.MetaBuildLoader.makeLoader(MetaBuildLoader.java:143)
	... 18 more
[error] [launcher] error during sbt launcher: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
```

--- a/main/src/main/java/sbt/internal/MetaBuildLoader.java
+++ b/main/src/main/java/sbt/internal/MetaBuildLoader.java
@@ -65,7 +65,7 @@ public final class MetaBuildLoader extends URLClassLoader {
    */
   public static MetaBuildLoader makeLoader(final AppProvider appProvider) throws IOException {
     final String jlineJars = "jline-?[0-9.]+-sbt-.*|jline-terminal(-(jna|jansi))?-[0-9.]+";
-    final String testInterfaceJars = "test-interface-[0-9.]+";
+    final String testInterfaceJars = "test-interface(-.*)?";
     final String compilerInterfaceJars = "compiler-interface(-.*)?";
     final String jansiJars = "jansi-[0-9.]+";
     final String jnaJars = "jna-(platform-)?[0-9.]+";

